addons/binding/org.openhab.binding.solarview/src/main/java/org/openhab/binding/solarview/internal/solarviewHandlerFactory.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.solarview.internal;
+
+import static org.openhab.binding.solarview.solarviewBindingConstants.*;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.openhab.binding.solarview.handler.solarviewHandler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link solarviewHandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Guenther Schreiner - Initial contribution
+ */
+public class solarviewHandlerFactory extends BaseThingHandlerFactory {
+
+    private static final Set<ThingTypeUID> SUPPORTED_THING_TYPES_UIDS = new HashSet<ThingTypeUID>(Arrays.asList(
+            THING_TYPE_ENERGY_PRODUCTION, THING_TYPE_PRODUCTION_INVERTER_ONE, THING_TYPE_PRODUCTION_INVERTER_TWO,
+            THING_TYPE_PRODUCTION_INVERTER_THREE, THING_TYPE_ENERGY_INJECTION, THING_TYPE_ENERGY_IMPORT));
+
+    private final Logger logger = LoggerFactory.getLogger(solarviewHandler.class);
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        logger.trace("supportsThingType({}) called.",thingTypeUID.getAsString());
```

Comment:
I believe the toString is correctly implemented there, you can just pass `thingTypeUID`